src/game_interpreter.cpp

```diff
@@ -2866,10 +2999,10 @@ bool Game_Interpreter::CommandKeyInputProc(lcf::rpg::EventCommand const& com) {
 		} else {
 			// For Rpg2k >=1.50
 			_keyinput.keys[Keys::eShift] = com.parameters[5] != 0;
-			_keyinput.keys[Keys::eDown] = param_size > 6 ? com.parameters[6] != 0 : false;
-			_keyinput.keys[Keys::eLeft] = param_size > 7 ? com.parameters[7] != 0 : false;
-			_keyinput.keys[Keys::eRight] = param_size > 8 ? com.parameters[8] != 0 : false;
-			_keyinput.keys[Keys::eUp] = param_size > 9 ? com.parameters[9] != 0 : false;
+			_keyinput.keys[Keys::eDown] = param_size > 6 && com.parameters[6] != 0;
```

Comment:
Maybe rewrite this as `check_param(X)` and do the size check there as well?

Reply:
found check_param vs check_key too confusing, added check_key where necessary instead